File: Computer science/Data structures and algorithms/Code/Data_Structure/src/com/fausto/DataStructure.java
package com.fausto;

/**
 * @author: fausto
 * @date: 2021/8/6 9:55
 * @description:
 */
public interface DataStructure {
	boolean isEmpty();
	
	boolean isFull();
	
	void addData(int data);
	
	int getData();
	
	void list();
	
	int peek();
}

File: Computer science/Data structures and algorithms/Code/Data_Structure/src/kotlin_way/DataStructure.kt
package kotlin_way

/**
 * @author: fausto
 * @date: 2021/10/5 12:02
 * @description: 数据结构接口
 */
open interface DataStructure<T> {

    //往结构中添加数据
    fun addData(data: T)

    //从结构中取出数据
    fun getData(): T

    //打印结构中所有数据
    fun list()

    //查看结构中第一个数据
    fun peek(): T
}